src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -79,7 +79,14 @@ private function registerHandlers(ContainerBuilder $container, array $busIds)
                     throw new RuntimeException(sprintf('Invalid handler service "%s": bus "%s" specified on the tag "%s" does not exist (known ones are: %s).', $serviceId, $tag['bus'], $this->handlerTag, implode(', ', $busIds)));
                 }
 
-                $className = $container->getDefinition($serviceId)->getClass();
+                $serviceIdDefinition = $container->getDefinition($serviceId);
```

Comment:
Can you put this logic (including the `while` loop) into a private method? I'd name it `getDefinitionClass(string $serviceId): string;`